### PR TITLE
Hardcases

### PR DESCRIPTION
**Pull request by shazbot194**
## About The Pull Request

This PR adds hardcases, starting with Engineering/Tools, Medical, Parts, and Ammo Cans. 

image

In most cases, hardcases are better boxes, but need to be opened and closed. While open they can be interacted like with how you normally do to boxes, but they take up a lot more space. You can Alt+click to open and close a hardcase.

Engineering/Tool cases hold Tools, Materials, parts, and almost anything engineering.
Medical cases hold most medical supplies.
Part cases can hold 20 components, parts, and boards.
Ammo cans can hold ammo, smaller ammo boxes, and magazine.

Right clicking you can also apply a sticker for orginization. 

There are both standard versions and scrap. There is no real difference between the two beyond aesthetics and scrap ones can be made by hand.

This PR also includes a small fix for AK stocks being able to be folded while in a bag. 

## Why It's Good For The Game

This helps allow more organization and storage both without needing lots of tables, crates, closets, ext to spread items out, while also not allowing it to be too easy to access. 

## Changelog
:cl:
add: Added Hardcases
fix: fixed AK folding stocks being foldable in bags.
/:cl:

### CODE REVIEW

**Comment by CeUvi** [comment]
image
these are just recolored Excel L-cells lmao

**Comment by ValoTheValo** [comment]
> image aren't these just recolored Excel L-cells lmao

They indeed are

**Comment by shazbot194** [comment]
Because the .dmi file is being weird, I swapped it for the original and will re-enable the one file I added. So at this current point, it is 1 for 1 with the main version.

File: code/datums/autolathe/containers.dm [hunk 1]
```diff
@@ -59,3 +59,27 @@
...
+	name = "Parts Hardcase"
+	build_path = /obj/item/storage/hcases/parts
+
+/datum/design/autolathe/container/hcase_med
+	name = "Parts Hardcase"
+	build_path = /obj/item/storage/hcases/med
+
+/datum/design/autolathe/container/hcase_engi
+	name = "Parts Hardcase"
+	build_path = /obj/item/storage/hcases/engi
```
[the remark this answers is not in the document]
Comment (shazbot194) [reply code/datums/autolathe/containers.dm:85]: They normally can only be made in factional disks, but I added them just for clarity

**Review by hyperioo — APPROVED** [review]

**Pull request merged**